### ISSUE TITLE
Getting Bytes kind as DAG-JSON from `ipfs dag get` returns wrong results 

### ISSUE THREAD

**Issue by d70-t** [ipfs/go-ipfs#8217]
#### Version information:
```
$ ipfs version --all
go-ipfs version: 0.9.0
Repo version: 11
System version: amd64/darwin
Golang version: go1.16.5
```

#### Description:
When running
```bash
ipfs block get bafyreigyko2ccjjk43hsr3iwnyz4kg4iqksp73xkwy5g2r2mbph6vbvzcq | hexdump
```

the result is:
```
0000000 a1 63 66 6f 6f 43 62 61 72                     
0000009
```

According to [cbor.me](http://cbor.me/?bytes=a163666f6f43626172), this is:
```
{"foo": h'626172'}
```
which is correct as I created the block to contain `{"foo": 'bar'}` as DAG-CBOR, where `bar` was intentionally encoded as bytes, not string.

When reading this back using `ipfs dag get`, I see the following:
```
$ ipfs dag get bafyreigyko2ccjjk43hsr3iwnyz4kg4iqksp73xkwy5g2r2mbph6vbvzcq
{"foo":"YmFy"}
```
which is not correct.

Interestingly, using `jsipfs`, the result is different, but also wrong:
```
$ jsipfs dag get bafyreigyko2ccjjk43hsr3iwnyz4kg4iqksp73xkwy5g2r2mbph6vbvzcq
{"foo":{"0":98,"1":97,"2":114}}
```

**Comment by Stebalien** [comment]
The issue in both of your cases is that go-ipfs does not (yet) support dag-json, just json. It returns the bytes as a base64 encoded string because it has nothing better to do.

**Comment by lidel** [comment]
This should be fixed in 0.10.0-rc1 – please give it a try and reopen if you see any issues
